ehitchhiking-ui/src/app/shared/components/blacklist/blacklist.component.ts

```diff
@@ -2,61 +2,71 @@ import {Component, OnInit} from '@angular/core';
 import {MatDialogRef} from '@angular/material';
 import {LoaderSize} from '@shared/enums/pre-loader-sizes';
 import {NoDataSize} from '@shared/enums/no-data-sizes';
-import {ApiService} from "@shared/services/api.service";
-import {URL_REGISTRY} from "@shared/constants/urlRegistry";
-import {User} from "@shared/models/user";
-import {BLACKLIST_DRIVERS, BLACKLIST_PASSENGERS, CUR_USER} from "@shared/components/blacklist/blacklist-users";
+import {User} from '@shared/models/user';
+import {BlackListApiService} from '@shared/api-services/black-list-api.service';
+import {CUR_USER} from '@shared/components/blacklist/blacklist-users';
 
 @Component({
 	selector: 'app-blacklist',
 	templateUrl: './blacklist.component.html',
 	styleUrls: ['./blacklist.component.sass'],
-  providers: [ApiService]
+	providers: [BlackListApiService],
 })
 export class BlacklistComponent implements OnInit {
-  blacklistDriverArray = [];
-  blacklistPassengerArray = [];
-  curUser: User = CUR_USER;
+	curUser: User = CUR_USER;
 	loaderSize: LoaderSize = LoaderSize.Large;
 	noDataSize: NoDataSize = NoDataSize.Small;
 	noDataMessage = 'No users!';
 	noDataIconName = 'accessibility';
-  loadingDrives = true;
-  loadingPassengers = true;
-	constructor(public dialogRef: MatDialogRef<BlacklistComponent>, public apiService: ApiService ) {}
+	loadingDrives = true;
+	loadingPassengers = false;
+	driversBlacklist: User[];
+	passengerBlacklist: User[];
 
-	ngOnInit() {
-    this.blacklistDriverArray = BLACKLIST_DRIVERS;
-    this.blacklistPassengerArray = BLACKLIST_PASSENGERS;
-		setTimeout(() => {
-      this.loadingDrives = false;
-		}, 1000);
-    setTimeout(() => {
-      this.loadingPassengers = false;
-    }, 2000);
+	constructor(public dialogRef: MatDialogRef<BlacklistComponent>, private apiBlacklistService: BlackListApiService) {}
 
-		this.apiService.doGet(URL_REGISTRY['blacklist.get'],false,{
-		  idDr: this.curUser.id
-    }).subscribe(data => {
-    console.log(data)
-    });
+	ngOnInit() {
+		console.log(this.apiBlacklistService.getBlockedDrivers({idDr: this.curUser.id}));
+		this.apiBlacklistService.getBlockedDrivers({idDr: this.curUser.id}).subscribe((data) => {
```

Comment:
this.loadingDrives = true;
this.apiBlacklistService.getBlockedDrivers({idDr: this.curUser.id}).subscribe((data) => {
			this.driversBlacklist = data;
			this.loadingDrives = false;
});

move to a separate function, this function will be helpful after removal of driver from black list

Reply:
done